Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -334,6 +347,97 @@ private Branch(Context context) {
 	}
 
 	/**
+	 *
+	 * <p>Creates and initialises a new  singleton instance of object of type {@link Branch} if
+	 * not existing.Application developers should not call this method directly.Instead extend
+	 * your Application class with {@link BranchLinkedApp} or use {@link BranchLinkedApp} as the
+	 * Application class in your manifest.
+	 * You need to set Branch key in your manifest</p>
+	 *
+	 * @see          <a href="https://github.com/BranchMetrics/Branch-Android-SDK/blob/05e234855f983ae022633eb01989adb05775532e/README.md#add-your-app-key-to-your-project">
+	 *                 Adding your app key to your project</a>
+	 *
+	 * @param application  Your {@link Application} instance.
+	 *
+	 */
+
+	public static void createInstance(Application application) {
+		isBranchAppInstantiated = application instanceof  BranchLinkApp;
+      /* Create and initialise a new instance of Branch. */
+		Branch createdInstance = getBranchInstance(application,true);
+
+		try {
+         /*Set an observer for ativity life cycle events*/
+			createdInstance.setActivityLifeCycleObserver(application);
+
+		}catch (NoSuchMethodError Ex){
+			//LifeCycleEvents are  available only from API level 14.
+			Log.w(TAG,BranchException.BRANCH_API_LVL_ERR_MSG);
+		}
+	}
+
+	/**
+	 * <p>Singleton method to return the pre-initialised object of the type {@link Branch}.
+	 * Make sure your app is  instantiating {@link BranchLinkedApp} before calling this method</p>
+	 *
+	 * @return An initialised singleton {@link Branch} object
+	 *
+	 * @throws BranchException Exception</br>
+	 *          1)If your {@link Application}  is not instance of {@link BranchLinkedApp} </br>
+	 *          2)If the minimum API level is below 14
+	 */
+	public static Branch getInstance() throws BranchException {
+		//Check if BranchLinkedApp is instantiated
+		if(branchReferral_ == null || isBranchAppInstantiated == false )
+			throw BranchException.getInstatiationException();
+
+			//Check if Activity life cycle callbacks are set
+		else {
+			if (branchReferral_.isActivityObserverInitialised() == false) {
+				throw BranchException.getAPILevelException();
+			}
+
+			else {
+				//Set branch Key with  the Live key
+				String branchKey = branchReferral_.prefHelper_.getBranchKey(true);
```

Comment:
Correct. Will remove the variable.
getKey() is used just to set the branch keys. prefHelper_.getBranchKey() method is actually setting the key internally. Is it a good idea to create a new method "setKey(isLive)" for setting the keys?